extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -0,0 +1,180 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @const */
+const MAX_ARTICLES = 2;
+
+/** @const */
+const SEPARATPOR_RECOS = 3;
+
+import {assertConfig} from './config';
+import {user} from '../../../src/log';
+import {tryParseJson} from '../../../src/json';
+import {CSS} from '../../../build/amp-document-recommendations-0.1.css';
+import {isJsonScriptTag} from '../../../src/dom';
+import {Services} from '../../../src/services';
+import {MultidocManager} from '../../../src/runtime';
+import {setStyle} from '../../../src/style';
+
+export class AmpDocumentRecommendations extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.element.classList.add('i-amp-document-recommendations');
+
+    // TODO(emarchiori): Consider using a service instead of singleton.
+    if (this.win.CONTENT_DISCOVERY) {
+      return;
+    }
+    this.win.CONTENT_DISCOVERY = this;
+
+    /** @private {?AmpDocumentRecommendationsConfig} */
+    this.config_;
+
+    /** @private {MultidocManager} */
+    this.multidocManager_;
+
+    /** @private {number} */
+    this.nextArticle_ = 0;
+  }
+
+  /** @override */
+  isLayoutSupported(unused) {
+    return true;
+  }
+
+  appendDivision_() {
+    const doc = this.win.document;
+    const topDivision = doc.createElement('div');
+    topDivision.classList.add('amp-document-recommendations-division');
+    this.element.appendChild(topDivision);
+  }
+
+  appendNextArticle_() {
+    if (this.nextArticle_ < MAX_ARTICLES &&
+        this.nextArticle_ < this.config_.recommendations.length) {
+      this.appendDivision_();
+      this.appendArticleLinks_(this.nextArticle_ + 1);
+
+      const next = this.config_.recommendations[this.nextArticle_];
+      this.nextArticle_++;
+
+      Services.xhrFor(this.win)
+          .fetchDocument(next.ampUrl, {})
+          .then(
+              doc => {this.attachShadowDoc_(doc);},
+              () => {});
+    }
+  }
+
+  /**
+   * @param {number} from
+   */
+  appendArticleLinks_(from) {
+    const doc = this.win.document;
+    let article = from;
+    const viewer = Services.viewerForDoc(this.getAmpDoc());
+
+    while (article < this.config_.recommendations.length &&
+        article - from < SEPARATPOR_RECOS) {
+      const next = this.config_.recommendations[article];
+      article++;
+
+      const articleHolder = doc.createElement('div');
```

Comment:
use `button` for A11Y, otherwise need to handle taxIndex and enter/space activation manually.

Reply:
Done.